keps/sig-apps/961-maxunavailable-for-statefulset/README.md

```diff
@@ -373,31 +837,62 @@ The struct in question is IntOrString.
 No
 
 ###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
-The controller-manager will see very negligible and almost un-notoceable increase in cpu usage.
+The controller-manager will see very negligible and almost un-noticeable increase in cpu usage.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No.
 
 ### Troubleshooting
 
 ###### How does this feature react if the API server and/or etcd is unavailable?
-The RollingUpdate will fail or will not be able to proceed if etcd or apiserver is unavailable and
+The RollingUpdate will fail or will not be able to proceed if etcd or API server is unavailable and
 hence this feature will also be not be able to be used.
 
 ###### What are other known failure modes?
-NA
+
+<!--
+For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+-->
+
+Generalized Max Unavailable Threshold Violations
+
+- The number of unavailable pods during a rolling update exceeds the specified maxUnavailable limit for many Stateful Sets.
+  - Detection: Monitor statefulset_unavailability_violation metric; a significant increase in this metric indicates the failure mode.
+  - Mitigations: Investigate the cause of excess unavailability and take corrective actions, such as adjusting the maxUnavailable parameter or troubleshooting problematic pods.
+  - Diagnostics: Increase logging verbosity and examine logs for messages indicating why pods are not becoming ready or why more pods than expected are being terminated.
+  - Testing: Implement e2e tests that simulate scenarios where pods fail to become ready and verify that the maxUnavailable limit is not exceeded.
+
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
 
 ## Implementation History
 
-- KEP Started on 1/1/2019
-- Implementation PR and UT by 8/30
+- 2019-01-01: KEP created.
+- 2019-08-30: PR Implemented with tests covered.
+- 2023-05-11: Bump to Beta.
 
 ## Drawbacks
 
-NA
+It's an opt-in feature so you can choose to use it or not, then no drawbacks I can think of.
```

Comment:
it can lead to increased application unavailability/outage during rollouts when this feature is used

Reply:
Adeed this